holoviews/core/spaces.py

```diff
@@ -788,6 +804,105 @@ def dynamic_redim(obj):
         return Dynamic(redimmed, shared_data=True, operation=dynamic_redim)
 
 
+    def collate(self):
+        """
+        Collation allows collapsing DynamicMaps with invalid nesting
+        hierarchies. This is particularly useful when defining
+        DynamicMaps returning an (Nd)Layout or GridSpace
+        type. Collating will split the DynamicMap into of individual
+        DynamicMaps. Note that the composite object has to be of
+        consistent length and types for this to work
+        correctly. Associating streams with specific viewables in the
+        returned container declare a stream_mapping on the DynamicMap
+        Callable during instantiation.
+        """
+        # Initialize
+        if self.last is not None:
+            pass
+        else:
+            self[self._initial_key()]
+
+        if isinstance(self.last, HoloMap):
+            # Get nested kdims and streams
+            streams = list(self.streams)
+            if isinstance(self.last, DynamicMap):
+                dimensions = [d(values=self.last.dimension_values(d.name))
+                              for d in self.last.kdims]
+                streams += self.last.streams
+                stream_kwargs = set()
+                for stream in streams:
+                    contents = set(stream.contents())
+                    if stream_kwargs & contents:
+                        raise KeyError('Cannot collate DynamicMaps with clashing '
+                                       'stream parameters.')
+            else:
+                dimensions = self.last.kdims
+            kdims = self.kdims+dimensions
+
+            # Define callback
+            def collation_cb(*args, **kwargs):
+                return self[args[:self.ndims]][args[self.ndims:]]
+            callback = Callable(collation_cb, inputs=[self])
+
+            return self.clone(shared_data=False, callback=callback,
+                              kdims=kdims, streams=streams)
+        elif isinstance(self.last, (Layout, NdLayout, GridSpace)):
+            # Expand Layout/NdLayout
+            from ..util import Dynamic
+            new_item = self.last.clone(shared_data=False)
```

Comment:
Based on the ``isinstance`` check, ``container`` might be a better name for this variable than ``new_item``...